src/server/system_services/bucket_server.js

```diff
@@ -1998,13 +2017,41 @@ function validate_replication(req) {
 }
 
 function normalize_replication(req) {
-    const replication_rules = req.rpc_params.replication_policy;
+    const replication_rules = req.rpc_params.replication_policy.rules;
 
-    const validated_replication = replication_rules.map(rule => {
+    const validated_replication_rules = replication_rules.map(rule => {
         const { destination_bucket } = rule;
         const dst_bucket = req.system.buckets_by_name && req.system.buckets_by_name[destination_bucket.unwrap()];
         return { ...rule, destination_bucket: dst_bucket._id };
     });
+
+    const validated_replication = { rules: validated_replication_rules };
+
+    if (req.rpc_params.replication_policy.log_replication_info) {
+        let validated_log_based_replication;
+        const logs_bucket = req.system.buckets_by_name &&
+            req.system.buckets_by_name[req.rpc_params.replication_policy.log_replication_info.logs_location.logs_bucket.unwrap()];
+        if (!logs_bucket) {
+            const source_bucket = find_bucket(req);
+            validated_log_based_replication = {
+                log_replication_info: { logs_location: { logs_bucket: source_bucket._id } }
+            };
+        } else {
+            validated_log_based_replication = {
+                log_replication_info: { logs_location: { logs_bucket: logs_bucket._id } }
+            };
+        }
+        if (req.rpc_params.replication_policy.log_replication_info.logs_location.subpath !== "") {
```

Comment:
why all this code is needed? 
I don't see why putting '/' for no subpath and '' for marker, keeping it empty won't work?

Reply:
putting '/' on the subpath it's just a default value if non provided, search the logs at the top of the bucket.
log marker can stay empty, planed to initiate it on the BG worker, and in the meanwhile put a placeholder in order to insert to the DB.

Reply:
Without '/' searching the logs will work? why do we need in the DB an empty marker? you can just check if it exists or not

Reply:
we could leave them empty and then when needed check existence and proceed as follows, I think that placing the default values now will save us time on always checking if they exist.